src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -2302,34 +2307,91 @@ protected override void HandleSetBreakpointsRequestAsync(IRequestResponder<SetBr
 
         protected override void HandleSetExceptionBreakpointsRequestAsync(IRequestResponder<SetExceptionBreakpointsArguments> responder)
         {
-            List<string> filter = responder.Arguments.Filters;
-            if (m_engineConfiguration.ExceptionSettings.Categories.Count > 0)
+            List<ExceptionFilterOptions> filterOptions = responder.Arguments.FilterOptions;
+            if (filterOptions != null && filterOptions.Count > 0)
             {
-                if (filter == null || filter.Count == 0)
+                foreach (ExceptionFilterOptions filterOption in filterOptions)
                 {
-                    SetAllExceptions(enum_EXCEPTION_STATE.EXCEPTION_STOP_SECOND_CHANCE);
-                }
-                else if (filter.Contains(ExceptionBreakpointFilter.Filter_All))
-                {
-                    enum_EXCEPTION_STATE state = enum_EXCEPTION_STATE.EXCEPTION_STOP_SECOND_CHANCE | enum_EXCEPTION_STATE.EXCEPTION_STOP_FIRST_CHANCE | enum_EXCEPTION_STATE.EXCEPTION_STOP_USER_FIRST_CHANCE;
+                    if (filterOption.FilterId == ExceptionBreakpointFilter.Filter_All)
+                    {
+                        if (!string.IsNullOrWhiteSpace(filterOption.Condition))
+                        {
+                            string[] conditions = filterOption.Condition.Split(',');
+                            ExceptionSettings.CategoryConfiguration category = m_engineConfiguration.ExceptionSettings.Categories.First(); // How to map filterOption.FilterId to Category Name / Guid.
```

Comment:
I was not sure how protocol handled the exception category for our engine's `EXCEPTION_INFO` structure.

Reply:
If we wanted to support this, `ExceptionBreakpointFilter` should have a category GUID that we used internally.

Reply:
Do I just map the "all" filterId to the GUID?

Reply:
If you wanted to fix this for real I would suggest:
* Get rid of the special understanding of 'all', and make any exception filter act like 'all' in that it sets the state to: `enum_EXCEPTION_STATE.EXCEPTION_STOP_SECOND_CHANCE | enum_EXCEPTION_STATE.EXCEPTION_STOP_FIRST_CHANCE | enum_EXCEPTION_STATE.EXCEPTION_STOP_USER_FIRST_CHANCE`
* Have each filter associated with a category

This would be enough to support signals in the future.

Reply:
I hope the latest commit is how you expected this fix to be.